### PR TITLE
Removed `T` prefix for types.

### LINKED ISSUE #7: Get rid of type prefixes

### PR DESCRIPTION
**Pull request by avahe-kellenberger**
Resolves #7

Let me know if there are any issues or changes needed to be made.

### CODE REVIEW

**Comment by SolitudeSF** [comment]
thats gonna break every package that uses x11

**Comment by avahe-kellenberger** [comment]
Just update the version to 2.0.0 and we should be fine

**Comment by Araq** [comment]
Well we can keep the old names like so:

```nim

type
  TFoo {.deprecated: "use Foo instead".} = Foo
```

**Comment by timotheecour** [comment]
Theres a better deprecated syntax for types (introduced for nimfix), it has techical advantages over this

**Comment by avahe-kellenberger** [comment]
The deprecatedAlias pragma? If we're using it, sure, I'll get on that today

**Comment by avahe-kellenberger** [comment]
Looked like `depreatedAlias` hasn't been added, so I went with Araq's suggestion.

File: examples/x11ex.nim [hunk 1]
```diff
@@ -42,7 +45,7 @@ proc create_window =
                                      PointerMotionMask or ExposureMask)
   discard XMapWindow(display, win)
 
-  wmDeleteMessage = XInternAtom(display, "WM_DELETE_WINDOW", false.TBool)
+  wmDeleteMessage = XInternAtom(display, "WM_DELETE_WINDOW", false.Bool)
```
Comment (Araq) [inline examples/x11ex.nim:48]: This must become ``X11Bool`` or similar, too subtle otherwise.

Comment (avahe-kellenberger) [reply examples/x11ex.nim:48]: @Araq How about XBool? To keep it the same length as before

**Comment by timotheecour** [comment]
@avahe-kellenberger 

> Looked like deprecatedAlias hasn't been added, so I went with Araq's suggestion.

https://github.com/nim-lang/RFCs/issues/214 hasn't been implemented yet, I'm talking about this:
```nim
{.deprecated: [TZipArchive: ZipArchive].}
```
it's better for several reasons:
* easier to automate in a tool like nimfix (nimfix can and should be revived)
* avoids need for a custom deprecation msg
* `type FooOld = FooNew` is not a true alias unlike `{.deprecated: [TZipArchive: ZipArchive].}` and has caveats

see https://github.com/nim-lang/RFCs/issues/214 
> why not use type FooOld {.deprecated.} = FooNew as was done in nim-lang/Nim#13908 ?

**Comment by avahe-kellenberger** [comment]
I get errors when using that syntax. Is it currently implemented or do we have docs about it?

**Comment by timotheecour** [comment]
see https://github.com/nim-lang/x11/pull/34 => no errors

btw, this PR breaks `nim c examples/x11ex.nim` 
so please add a test to x11.nimble to make sure nothing breaks; it can be as simple as:

```nim
task test, "test":
  exec "nim c examples/x11ex.nim"
  exec "nim c examples/xshmex.nim"
```
which is better than nothing

**Comment by avahe-kellenberger** [comment]
Thanks for the example! I'll make those changes. Also, it still compiles and runs for me - what is breaking for you?

**Comment by timotheecour** [comment]
```
git rev-parse HEAD
5e3373ba49b1ab8a08ca861507068a10d80ce97a
nimble develop # make sure you're using your stuff
nim c examples/x11ex.nim
/Users/timothee/git_clone/nim/temp/x11/examples/x11ex.nim(22, 1) template/generic instantiation from here
/Users/timothee/git_clone/nim/temp/x11/examples/x11ex.nim(48, 67) Error: undeclared field: 'Bool' for type system.bool [declared in /Users/timothee/git_clone/nim/Nim_devel/lib/system/basic_types.nim(15, 3)]
    wmDeleteMessage = XInternAtom(display, "WM_DELETE_WINDOW", false.Bool)
```

**Comment by avahe-kellenberger** [comment]
Fixed - hopefully this should be good now.

**Comment by timotheecour** [comment]
I've reviewed almost all of it and it looks good; I'm assuming you've used a script to make this PR, if so if reviewer runs same script and gets to same result it'd give even more confidence that everything is a-ok
(nimfix should grow this kind of tool for the benefit of all future refactorings...)

**Comment by avahe-kellenberger** [comment]
Just by hand/with vim. I've looked through everything twice, but it's always good to review

**Pull request merged**